ultravox/inference/base.py

```diff
@@ -45,8 +50,51 @@ def infer_stream(
         sample: datasets.VoiceSample,
         max_tokens: Optional[int] = None,
         temperature: Optional[float] = None,
+        past_key_values: Optional[Union[Tuple, transformers.cache_utils.Cache]] = None,
     ) -> InferenceGenerator:
         """Streaming polyfill, if not supported directly in derived classes."""
-        output = self.infer(sample, max_tokens, temperature)
+        output = self.infer(sample, max_tokens, temperature, past_key_values)
         yield InferenceChunk(output.text)
         yield InferenceStats(output.input_tokens, output.output_tokens)
+
+
+class History:
+    def __init__(self, audio_token_replacement: str = "<|eot_token|>"):
+        self.audio_token_replacement: str = audio_token_replacement
+        self.audio_placeholder = "<|audio|>"
+        self.messages: List[Dict[str, str]] = []
+        self.key_values: Optional[Union[Tuple, transformers.cache_utils.Cache]] = None
+
+    def add_message(self, message: Dict[str, str], audio_token_len: int):
```

Comment:
I'd prefer to keep the same interface here as in the model class, where we specify lists of audio insert position and audio len, rather than using a magic token string. Alternatively we could make the interface be a List of `MessagePart`, where each part is either a string or an audio chunk (similar to the OpenAI API)